diff -urN a/epgrec/reclib.php b/epgrec/reclib.php
--- a/epgrec/reclib.php	2016-04-11 23:00:00.000000000 +0900
+++ b/epgrec/reclib.php	2022-11-24 01:20:28.815913173 +0900
@@ -283,16 +283,16 @@
 }
 
 function get_ipckey( $id ){
-/*
+
 	if( !file_exists( FTOK_KEY ) ){
 		$handle = fopen( FTOK_KEY, 'w' );
 		fwrite( $handle, 'a' );
 		fclose( $handle );
 //		exec( 'sync' );
 	}
-	return ftok( FTOK_KEY, $id );	// ftok()は、仕様上で唯一性を担保できないバグあり
-*/
-	return $id;
+	return ftok( FTOK_KEY, chr($id) );	// ftok()は、仕様上で唯一性を担保できないバグあり
+
+//	return $id;
 }
 
 function run_user_regulate(){
